### ISSUE TITLE
Add Filecoin custom multihashes?

### ISSUE THREAD

**Issue by jimpick** [multiformats/multihash#129]
Filecoin defines some new multihash codes here:

https://github.com/filecoin-project/go-fil-commcid/blob/05624bdb669c908632a93b826c25e9b579237d65/commcid.go#L20

Should the master list of multihash codes include there?

I tried plugging a PieceCID `bafk4chzaammouedlmmvekovrg4ema7lvuj5fzfqdrcycg5vfnrioxxjmo4ta` into https://cid.ipfs.io/ and it complained "multihash unknown function code: 0xfc1".

https://cid.ipfs.io/#bafk4chzaammouedlmmvekovrg4ema7lvuj5fzfqdrcycg5vfnrioxxjmo4ta

Also, the `ipfs cid` tool can't identify the CIDs:

```
$ ipfs cid format bafk4chzaammouedlmmvekovrg4ema7lvuj5fzfqdrcycg5vfnrioxxjmo4ta -f "%b CIDv%V %h"
base32 CIDv1 hash?4033
```

**Comment by ribasushi** [comment]
@jimpick thank you for surfacing this, it actually got lost in my set of notes 🙀 after @rvagg alerted me to it last week
TLDR is this:
- The way go-cid and friends work makes "placeholders" really awkward
  => `0xfc{{X}}` was chosen to work around that knowing that these are not the final values
- The final values were settled after a very long back and forth, and are now: https://github.com/multiformats/multicodec/pull/161#issuecomment-627152617
```
sha2-256-trunc254-padded: 0x1012
poseidon-bls12_381-a2-fc1: 0xb401
poseidon-bls12_381-a2-fc1-sc: 0xb402
fil-commitment-unsealed: 0xf101
fil-commitment-sealed: 0xf102
```

My understanding is that the plan was to smply fix https://github.com/filecoin-project/go-fil-commcid/blob/master/commcid.go#L18-L24 since these bytes are not anywhere on-chain.

Can you confirm?

**Comment by jimpick** [comment]
I believe the PieceCIDs are on chain as they are part of the deal proposal.

**Comment by ribasushi** [comment]
@jimpick I gathered more feedback/history on this, and apparantley the plan *was* to change the IDs on-chain before mainnet. I opened a PR to start things moving in this direction: https://github.com/filecoin-project/go-fil-commcid/pull/2#issuecomment-652969693

We will close this ( https://github.com/multiformats/multihash/issues/129 ) issue once the other PR takes off.

**Comment by ribasushi** [comment]
@jimpick the new CIDs are now included in https://github.com/filecoin-project/lotus/tree/next
The cid explorer should display the new `commX` cids correctly on next chain reset

/cc @mikeal @rvagg

**Comment by lidel** [comment]
Hey folks, where can I find sample CIDs produced by Filecoin?
Want to make sure our GUIs work fine when someone pastes them.

**Comment by ribasushi** [comment]
@lidel good catch, it doesn't work 🙀 

Not sure where to open an issue for `https://cid.ipfs.io/` : just c/p the below testcases where they belong.

[baga6ea4seaqggjjfh7whhdoxvhrix6jbcgobmdyhajcimfn33iedcp3kr23gruq](https://cid.ipfs.io/#baga6ea4seaqggjjfh7whhdoxvhrix6jbcgobmdyhajcimfn33iedcp3kr23gruq) ( [raw bytes](https://gchq.github.io/CyberChef/#recipe=Drop_bytes(1,-1,false)From_Base32('a-z2-7%3D',false)To_Hex('Space',0)&input=YmFnYTZlYTRzZWFxZ2dqamZoN3doaGRveHZocml4NmpiY2dvYm1keWhhamNpbWZuMzNpZWRjcDNrcjIzZ3J1cQ) )
[baga6ea4seaqidbk23bub2dmg2hur4aawpe44wzuu2lccflgsbcqaokjzjb7wtgi](https://cid.ipfs.io/#baga6ea4seaqidbk23bub2dmg2hur4aawpe44wzuu2lccflgsbcqaokjzjb7wtgi) ( [raw bytes](https://gchq.github.io/CyberChef/#recipe=Drop_bytes(1,-1,false)From_Base32('a-z2-7%3D',false)To_Hex('Space',0)&input=YmFnYTZlYTRzZWFxaWRiazIzYnViMmRtZzJodXI0YWF3cGU0NHd6dXUybGNjZmxnc2JjcWFva2p6amI3d3RnaQ) )
[bagboea4b5abcax5zbow3g7cyeg3nsvjguqnjkbdibnhhzc3whinr2sousvoijbrb](https://cid.ipfs.io/#bagboea4b5abcax5zbow3g7cyeg3nsvjguqnjkbdibnhhzc3whinr2sousvoijbrb) ( [raw bytes](https://gchq.github.io/CyberChef/#recipe=Drop_bytes(1,-1,false)From_Base32('a-z2-7%3D',false)To_Hex('Space',0)&input=YmFnYm9lYTRiNWFiY2F4NXpib3czZzdjeWVnM25zdmpndXFuamtiZGlibmhoemMzd2hpbnIyc291c3ZvaWpicmI) )
[bagboea4b5abcb245dcsepbaelwd7hrt46itun2mvv5nckntzkg5kf73m2ry4ja7r](https://cid.ipfs.io/#bagboea4b5abcb245dcsepbaelwd7hrt46itun2mvv5nckntzkg5kf73m2ry4ja7r) ( [raw bytes](https://gchq.github.io/CyberChef/#recipe=Drop_bytes(1,-1,false)From_Base32('a-z2-7%3D',false)To_Hex('Space',0)&input=YmFnYm9lYTRiNWFiY2IyNDVkY3NlcGJhZWx3ZDdocnQ0Nml0dW4ybXZ2NW5ja250emtnNWtmNzNtMnJ5NGphN3I) )

**Comment by vmx** [comment]
The source for https://cid.ipfs.io/ is at https://github.com/multiformats/cid-utils-website.

**Comment by lidel** [comment]
Done:

```
updated TXT _dnslink.cid.ipfs.io to dnslink=/ipfs/bafybeifuvxkoxxjrjv47qvog26a4iwglmeq4fnnmqxrikc4g4u2ob7jzre
```

When DNSLink update is applied, 

- https://cid.ipfs.io/#bagboea4b5abcb245dcsepbaelwd7hrt46itun2mvv5nckntzkg5kf73m2ry4ja7r 

should work the same as 

- https://bafybeifuvxkoxxjrjv47qvog26a4iwglmeq4fnnmqxrikc4g4u2ob7jzre.ipfs.dweb.link/#bagboea4b5abcb245dcsepbaelwd7hrt46itun2mvv5nckntzkg5kf73m2ry4ja7r
